tools/wake/main.cpp

```diff
@@ -622,57 +639,64 @@ int main(int argc, char **argv) {
   std::vector<ExternalFile> wakefiles;
   wakefiles.reserve(wakefilenames.size());
 
-  bool alerted_slow_cache = false;
-  // While the slow cache alert is helpful, its also flakey.
-  // In order to support automated flows better we only emit it when
-  // a terminal is being used, which is a good indicator of a human
-  // using wake rather than an automated flow.
-  bool is_stdout_tty = isatty(1);
-
-  for (size_t i = 0; i < wakefilenames.size(); i++) {
-    auto &wakefile = wakefilenames[i];
-
-    auto now = std::chrono::steady_clock::now();
-    if (!clo.quiet && is_stdout_tty &&
-        std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count() > 1000) {
-      std::cout << "Scanning " << i + 1 << "/" << wakefilenames.size() << " wake files.\r"
-                << std::flush;
-      start = now;
-      alerted_slow_cache = true;
-    }
+  {
+    // While the slow cache alert is helpful, its also flakey.
+    // In order to support automated flows better we only emit it when
+    // a terminal is being used, which is a good indicator of a human
+    // using wake rather than an automated flow.
+    bool is_stdout_tty = isatty(1);
+    bool alerted_slow_cache = false;
+
+    struct timeval starttv, stop;
+    gettimeofday(&starttv, 0);
+
+    for (size_t i = 0; i < wakefilenames.size(); i++) {
+      auto &wakefile = wakefilenames[i];
+
+      auto now = std::chrono::steady_clock::now();
+      if (!clo.quiet && is_stdout_tty &&
+          std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count() > 1000) {
```

Comment:
nit: its very strange that we're using both chrono and gettimeofday

Reply:
yeah, :/ this was lifted from some other profiling code we have. I can see about just using `chrono`